### PR TITLE
Add types for @paypal/paypal-js

### PR DESCRIPTION
**Pull request by ReazerDev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/paypal__paypal-js/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (amcasey) [inline types/paypal__paypal-js/tsconfig.json:9]: Why disable `strictNullChecks`?

File: types/paypal__paypal-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for @paypal/paypal-js 1.0
+// Project: https://github.com/paypal/paypal-js
+// Definitions by: ReazerDev <https://github.com/ReazerDev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// Type definitions for npm package @paypal/paypal-js
+
+/// <reference types="node" />
```
Comment (amcasey) [inline types/paypal__paypal-js/index.d.ts:8]: Where is this used?

File: types/paypal__paypal-js/tsconfig.json (hunk 1 above)
Comment (amcasey) [inline types/paypal__paypal-js/tsconfig.json:9]: Why disable `strictNullChecks`?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/paypal__paypal-js/index.d.ts (hunk 2 above)
Comment (ReazerDev) [reply types/paypal__paypal-js/index.d.ts:8]: `// Type definitions for npm package @paypal/paypal-js` ?

Comment (ReazerDev) [reply types/paypal__paypal-js/index.d.ts:8]: or `/// <reference types="node" />` ?

Comment (amcasey) [reply types/paypal__paypal-js/index.d.ts:8]: GH includes the first three lines for context - the comment refers to the last line (in this case, `/// < reference`).

Comment (ReazerDev) [reply types/paypal__paypal-js/index.d.ts:8]: Saw it in another definition lol Just took a look at some others and saw, that it's not there haha Gonna remove it

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by ReazerDev** [comment]
Ready to merge

**Pull request merged**